docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -90,4 +97,139 @@ hs.refreshIndex("empIndex", "incremental")
 TODO
 
 ### Hybrid Scan
-TODO
+
+Hybrid Scan enables to utilize existing index data along with newly appended source files or
+deleted existing files, without explicit refresh operation. For an index with appended source data files,
+HybridScan changes the query plan to shuffle new data on-the-fly and merge it with index records.
+For an index with deleted source data files, Hyperspace also modifies the plan to exclude the rows from deleted files
+in the index data. This requires enabling lineage for the index at its creation time.
+
+Currently, HybridScan is disabled by default. You can check the [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/)
+page to see how it can be enabled.
+
+In the current version (0.3), if a dataset has many deleted source data files, query performance
+could degrade when Hybrid Scan is enforcing deletes at the query runtime. 
+Hyperspace provides two different configurations to tune this behavior. 
+You can use them to enable supporting deletes during Hybrid Scan and determine when it should be applied,
+depending on the total number of deleted source data files.
+
+#### Append-only dataset
+
+If your dataset is append-only dataset, you can use Hybrid Scan for appended files only.
+In this case, Hyperspace will not pick an index with some deleted source data file(s) for Hybrid Scan.
```

Comment:
Should we also mention that lineage is not required in this scenario as a side note?